kvbc/include/categorization/updates.h

```diff
@@ -0,0 +1,116 @@
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the
+// "License").  You may not use this product except in compliance with the
+// Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include "assertUtils.hpp"
+#include "kv_types.hpp"
+
+#include <ctime>
+#include <map>
+#include <optional>
+#include <set>
+#include <string>
+#include <variant>
+
+// Categorized key-value updates for KVBC blocks. Every category supports different properties and functionalities.
+// Note1: Empty category IDs are invalid and not supported.
+// Note2: Using the same category ID for different category types is an error.
+namespace concord::kvbc::categorization {
+
+// Updates for multiple categories.
+// Persists key-values directly in the underlying key-value store. All key-values are marked stale during the update
+// itself. Explicit deletes are not supported.
+struct CategorizedKeyValueUpdate {
+  struct CategorizedValue {
```

Comment:
I find this name confusing, maybe we can think if more meaningful name

Reply:
I agree it might not be ideal. Any suggestions?

Another thing we can do is to maybe avoid such an update type and require users to update a key across (potentially) multiple groups explicitly.

Reply:
Renamed to `SharedKeyValueUpdates` and `SharedValue` as suggested by @cloudnoize in an offline discussion.